### PR TITLE
Transaction CRUD

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- Added get function for transaction
- Added get by id function
- Added custom query function
- Added a extract function to extract the fields

### CODE REVIEW

File: routes/transaction_routes.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@router.get("/transaction/{appointment_id}/{transaction_id}")
+@permission_required("TRANSACTION", "READ")
+async def get_transaction_by_id(
+    request: Request,
+    appointment_id: str,
+    transaction_id: str
+):
+    return TransactionClient.get_by_id(appointment_id, transaction_id)
+
+@router.get("/transaction/detail")
```
Comment (tamsud) [inline routes/transaction_routes.py:29]: can you move this to get all itself

File: controller/transaction_controller.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            json_data = {
+                "data": transaction_data
+            }
+        else:
+            raise HTTPException("provide a valid query to download the file")
+        data = json_data['data']
+        df = pd.DataFrame(data)
+        timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
+        filename = f"{CSV_FILE_NAME}_{timestamp}.csv"
+        csv_output = BytesIO()
```
Comment (tamsud) [inline controller/transaction_controller.py:281]: Why are we using ByteIO for csv?

File: controller/transaction_controller.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        df.to_csv(csv_output, index=False)
+        csv_output.seek(0)
+        with open(filename, "wb") as f:
+            f.write(csv_output.getvalue())
```
Comment (tamsud) [inline controller/transaction_controller.py:285]: why we need to store the file?

File: controller/transaction_controller.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        file_content = csv_output.getvalue()
+        try:
+            response = StreamingResponse(
+                BytesIO(file_content),
+                media_type="text/csv",
+                headers={"Content-Disposition": f"attachment; filename={filename}"}
+            )
+            return response
+        finally:
+            TransactionClient.delete_file(filename)
```
Comment (tamsud) [inline controller/transaction_controller.py:296]: why we need to delete the csv?

**Review by tamsud — CHANGES_REQUESTED** [review]

File: routes/transaction_routes.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+from fastapi import APIRouter, Request, Query
+
+from utils.common_utils import permission_required
+from controller.transaction_controller import TransactionClient
+
+router = APIRouter()
+logger = logging.getLogger("log")
+
+@router.get("/transaction")
+@permission_required("TRANSACTION", "READ")
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply routes/transaction_routes.py]: changed to ALL_READ permission

File: controller/transaction_controller.py (hunk 2 above)
Comment (MonishaSivanathan) [reply controller/transaction_controller.py:285]: To store the file in memory and and remove

File: controller/transaction_controller.py (hunk 3 above)
Comment (MonishaSivanathan) [reply controller/transaction_controller.py]: without storing cant return

File: controller/transaction_controller.py (hunk 4 above)
Comment (tamsud) [reply controller/transaction_controller.py:296]: why we need to store and delete incase if its already in memory

File: controller/transaction_controller.py (hunk 2 above)
Comment (MonishaSivanathan) [reply controller/transaction_controller.py:285]: @tamsud i have removed the function and im not storing any where, im generating the on the flow and return

File: controller/transaction_controller.py (hunk 3 above)
Comment (MonishaSivanathan) [reply controller/transaction_controller.py]: no @tamsud im not storing

File: controller/transaction_controller.py (hunk 4 above)
Comment (MonishaSivanathan) [reply controller/transaction_controller.py:296]: i have changed tamil, im not storing

**Review by tamsud — APPROVED** [review]

**Pull request merged**